docs/validators.md

```diff
@@ -0,0 +1,21 @@
+Validators
+==========
+
+Validators allow you to validate or transform command arguments just before execution.
+
+Knack supports both command-level and argument-level validators.
+
+If a command has a command-level validator, then any argument-level validators that would ordinarily be applied are ignored.
+
+i.e. A command can have at most command validator or many argument level validators.
+
+**Command-level Validators**
+
+Command-level validators can operate on any arguments on the command. This is useful when you want a validator to make use of more than a single argument.
+
+**Argument-level Validators**
+
+Argument-level validators should only operate on a single argument.
+
+The order argument-level validators are executed in is not guaranteed so don't use an arguments that make use of multiple arguments.
```

Comment:
Yeah, this isn't excactly true. For example, you could have a single validator that makes use of `subnet_name` and `vnet_name` for example.  It's totally fine to have this in a single validator, just as long as some OTHER validator doesn't rely on them.